### ISSUE TITLE
Machine Learning notebook

### ISSUE THREAD

**Issue by Kk1984up** [Kk1984up/deeplearning2020#4]
note：
输入大小 Win X Hin X Din
卷积神经网络超参数，需要记住，一般需要指定的超参数，fiter个数（K输出的特征通道数channels），fiter尺寸大小（F）Height，width，滑动步长stides，边界填充padding。输出的计算公式
Wout = （Win -F+2P）/S   +1
Hout =  (Hin -F +2P)/S +1
Dout  = K
需要学习的参数param = （K X K +1）X Din X Dout
池化操作是不产生参数的，只是一个下采样过程。

day1:安装tensorflow
使用anaconda命令行中直接输入： pip install tensorflow-cpu==2.3.0 -i https://pypi.douban.com/simple/
上述安装命令行的意思是使用豆瓣源安装2.3.0版本的CPU tensorflow，豆瓣源目前是国内最快的安装镜像源，非常快。

**Comment by Kk1984up** [comment]
Tensorflow/Tensorflow-cpu/Tensorflow-gpu的区别
在tensorflow 1.x 只有tensorflow和tensorflow-gpu

>只有CPU时，两种都是CPU运行，

> 有GPU且安装cuda和cudnn，tensorflow-gpu版本是GPU运行，

> tensorflow 2.x不再区分是否gpu，当检测到gpu并安装cuda后，自动调用gpu。

但是，有些人不需要或没有gpu，gpu适配对这部分群体是浪费的（占用不必要的资源），于是有了tensorflow-cpu，我们可以理解其为cpu only版本

（综上，也可以理解为：tensorflow==1.x对应tensorflow-cpu==2.x，tensorflow-gpu==1.x对应tensorflow==2.x）

**Comment by Kk1984up** [comment]
day2：BN层的含义（BatchNormalization）批归一化
基本思想是，让每个隐藏节点的激活输入分布固定下来?解决内部协方差偏移问题
BN层是在前一个特征层的输出到激活函数层之间，基本原理就是将输入特征层的参数经过白化操作（输入值减去均值，然后除以标准差，将输入值的分布映射到均值为0，标准差为1 的正态分布），sigmoid激活函数是一个sigmoid = 1/(1+e（-z）)，输出为0-1之间，代表着概率，激活函数的导数在靠近-1.1之间比较大，往正负无穷之后，激活函数的导数变小，趋于梯度饱和区，不利于反向传播的收敛。
综上所示，bn层的最主要目的是为了解决深度神经网络出现梯度饱和的问题，

同样的残差网络也是为了解决更深的层出现的梯度消失和难以收敛的问题，创造性的提出shortcut层，
